intermediate_source/pipeline_tutorial.py

```diff
@@ -266,6 +266,7 @@ def get_batch(source, i):
     world_size=1,
     rpc_backend_options=rpc.TensorPipeRpcBackendOptions(
         init_method="file://{}".format(tmpfile.name),
+        _transports=["ibv", "uv"],
```

Comment:
Looks like one of the build workers is segfaulting after this change during RPC init: https://app.circleci.com/pipelines/github/pytorch/tutorials/3607/workflows/06cd1387-e1f0-43c2-9bca-c5ddd168ac6e/jobs/67531. I was wondering if you knew of any related issues while setting these options?

Looks like the default docker container doesn't have ptrace enabled so I can't get the stack trace from gdb. And If I try to build a custom docker container with ptrace enabled, the issue doesn't reproduce :(

Reply:
I tried `_transports=["uv"],` and it segfaults as well.

Reply:
It's not a segfault, it's an uncaught exception raised after some syscall returned EINVAL. It probably doesn't come from the uv backend, it could come from some channel. Without the stack trace of where this is raised it's gonna be hard to debug it. I tried re-running the failed job but I don't have the permission, possibly because it's a branch from your fork? Is there any way you can give me access so I can poke this issue a bit?

Reply:
Right its not a segfault, my bad. As I mentioned above, I couldn't find a way of getting a stack trace since whenever I built a docker container with gdb, the issue wouldn't reproduce.

@brianjo Is there a way for @lw to get permission to "Rerun Job with SSH" for jobs on my PR? 

If not, I guess one way you can reproduce this is to clone my PR and just submit a dummy PR from your account and then you probably can SSH into the appropriate CI job.

Reply:
I cloned and re-submitted this PR (#1409), and I was able to SSH into the job and I managed to attach GDB and get a trace of the exception (it kinda works if you launch GDB from outside the container and attach it to a process inside the container). I got [this](https://gist.github.com/lw/14ecb77392cdc186361d7ce3c172728e) which shows the problem comes from the CMA channel. To be precise, it happens when CMA fails to initialize and thus is left in an incomplete state, which causes problems at destruction.

I've fixed this in https://github.com/pytorch/tensorpipe/pull/313, which I'm landing now and which I'll cherry-pick into v1.8.1. In the meantime, to work around this, you can either set up Docker so that CMA does not fail its initialization (i.e., by passing `--security-opt seccomp=unconfined` to `docker run`), or by using the `_channels` option of the TP agent to exclude the `cma` channel from being used.